python/cuml/test/test_tsne.py

```diff
@@ -0,0 +1,130 @@
+
+
+from cuml.manifold import TSNE
+
+from sklearn.manifold.t_sne import trustworthiness
+from sklearn import datasets
+import pandas as pd
+import numpy as np
+import cudf
+import pytest
+
+
+def unit_param(*args, **kwargs):
+    return pytest.param(*args, **kwargs, marks=pytest.mark.unit)
+
+
+def quality_param(*args, **kwargs):
+    return pytest.param(*args, **kwargs, marks=pytest.mark.quality)
+
+
+def stress_param(*args, **kwargs):
+    return pytest.param(*args, **kwargs, marks=pytest.mark.stress)
+
+
+@pytest.mark.parametrize('nrows', [unit_param(30), quality_param(5000),
+                         stress_param(500000)])
+@pytest.mark.parametrize('n_feats', [unit_param(10), quality_param(100),
+                         stress_param(1000)])
+def test_blobs_cluster(nrows, n_feats):
+    X, _ = datasets.make_blobs(
+        n_samples=nrows, n_features=n_feats, centers=5, random_state=0)
+    Y = TSNE(verbose=False).fit_transform(X)
+
+    trust = trustworthiness(X, Y)
+    print("Trust = ", trust)
+    if trust < 0.97:
+        if trust < 0.95:
+            assert trust > 0.9
+    del X, Y
+
+
+def test_tsne():
+    """
+    This tests how TSNE handles a lot of input data across time.
+    (1) cuDF DataFrames are passed input
+    (2) Numpy arrays are passed in
+    (3) Params are changed in the TSNE class
+    (4) The class gets re-used across time
+    (5) Trustworthiness is checked
+    (6) Tests NAN in TSNE output for learning rate explosions
+    (7) Tests verbosity
+    """
+    from sklearn.datasets import load_digits as data1
+    from sklearn.datasets import load_boston as data2
+    from sklearn.datasets import load_iris as data3
+    from sklearn.datasets import load_breast_cancer as data4
+    from sklearn.datasets import load_diabetes as data5
+
+    X1 = data1().data
```

Comment:
Please parameterize these as function arguments. This will clean up this implementation and make it more reusable.

Reply:
Good idea :) Will fix